# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/AbstractRange.cpp
# Commit: 4d69d8456005
# Full Hash: 4d69d8456005efeb2ad4581ff00120e63c0bbdbc
# Author: Serban Stanca <sstanca@mozilla.com>
# Date: 2025-05-09 04:15:00
# Regressor Bug: 1932150
# File Overlap Count: 2
# Description:
#   Revert "Bug 1932150 - Fix some inconsistency about indicating whether the selection is allowed to cross the shadow boundary r=jjaschke,dom-core"
#   
#   This reverts commit f0836d70a70a1dbc6d74032586a1d5658243ea97.
# ==============================================================================

diff -r 48fc3f91f020 -r 4d69d8456005 dom/base/AbstractRange.cpp
--- a/dom/base/AbstractRange.cpp	Thu May 08 22:53:07 2025 +0300
+++ b/dom/base/AbstractRange.cpp	Thu May 08 22:53:07 2025 +0300
@@ -249,10 +249,14 @@
   if (!mIsPositioned) {
     return nullptr;
   }
-  nsINode* startContainer = ShadowDOMSelectionHelpers::GetStartContainer(
-      this, aAllowCrossShadowBoundary);
-  nsINode* endContainer = ShadowDOMSelectionHelpers::GetEndContainer(
-      this, aAllowCrossShadowBoundary);
+  nsINode* startContainer =
+      aAllowCrossShadowBoundary == AllowRangeCrossShadowBoundary::Yes
+          ? GetMayCrossShadowBoundaryStartContainer()
+          : GetStartContainer();
+  nsINode* endContainer =
+      aAllowCrossShadowBoundary == AllowRangeCrossShadowBoundary::Yes
+          ? GetMayCrossShadowBoundaryEndContainer()
+          : GetEndContainer();
 
   if (MayCrossShadowBoundary() &&
       aAllowCrossShadowBoundary == AllowRangeCrossShadowBoundary::Yes) {